streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -843,7 +843,7 @@ public KafkaStreams(final Topology topology,
     public KafkaStreams(final Topology topology,
                         final StreamsConfig applicationConfigs,
                         final KafkaClientSupplier clientSupplier) {
-        this(new TopologyMetadata(topology.internalTopologyBuilder, applicationConfigs), applicationConfigs, clientSupplier);
+        this(topology, new InternalStreamsConfig(applicationConfigs), clientSupplier, Time.SYSTEM);
```

Comment:
Some side cleanup on constructor methods -- no public change -- just to make the flow simpler